### ISSUE TITLE
VTK cannot be installed (erroneous check within formula)

### ISSUE THREAD

**Issue by Argon-** [Homebrew/homebrew-core#21678]
# Please always follow these steps:
- [x] Confirmed this is a problem with `brew install`ing one, specific Homebrew/homebrew-core formula (not cask or tap) and not every time you run `brew`? If it's a general `brew` problem please file this issue at https://github.com/Homebrew/brew/issues/new. If it's a `brew cask` problem please file this issue at https://github.com/caskroom/homebrew-cask/issues/new. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed all issues and retried your prior step?
- [x] Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

gist-logs: https://gist.github.com/9cc8504cb37b25dec20afd73fc290f3a
formula in question: https://github.com/Homebrew/homebrew-core/blob/master/Formula/vtk.rb

I tried to install VTK using: `brew install vtk --without-python --with-python3 --with-qt`
This fails due to checks within the formula itself. To be precise, it fails here: https://github.com/Homebrew/homebrew-core/blob/master/Formula/vtk.rb#L85
Output:

    → brew install vtk --without-python --with-python3 --with-qt
    ==> Using Homebrew-provided fortran compiler.
    This may be changed by setting the FC environment variable.
    ==> Downloading https://www.vtk.org/files/release/8.0/VTK-8.0.1.tar.gz
    Already downloaded: /Users/Julian/Library/Caches/Homebrew/vtk-8.0.1.tar.gz
    Failed to import the site module
    Your PYTHONPATH points to a site-packages dir for Python 2.x but you are running Python 3.x!
         PYTHONPATH is currently: "/usr/local/lib/python2.7/site-packages"
         You should `unset PYTHONPATH` to fix this.
    Failed to import the site module
    Your PYTHONPATH points to a site-packages dir for Python 2.x but you are running Python 3.x!
         PYTHONPATH is currently: "/usr/local/lib/python2.7/site-packages"
         You should `unset PYTHONPATH` to fix this.
    Failed to import the site module
    Your PYTHONPATH points to a site-packages dir for Python 2.x but you are running Python 3.x!
         PYTHONPATH is currently: "/usr/local/lib/python2.7/site-packages"
         You should `unset PYTHONPATH` to fix this.
    Error: No libpythonX.Y.{dylib|a} file found!

You can see the error in the last line.
I tried to reconstruct what it is (probably) doing in the formula _by executing the respecting commands in my terminal_:
* L67: `which python3` -> `/usr/local/bin/python3`
* L69: `#{python_executable} -c 'import sys;print(sys.prefix)'` -> `/usr/local/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6`
* L78: obviously, when we arrive in L85, this check must fail. However, using the above strings, the probed file does exist: `/usr/local/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/Python`. The check for `/usr/local/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/libpython3.6.dylib` should also work, the `*.a` does obviously not exist though.

It should work and I can't really think of a reason why it doesn't detect these clearly existing files, except when the formula gets some basic information (such as python version) wrong in the first place. But I can't really make sense of how it is determined which version to use? Is there magic going on by specifying `--with-python3`?
I also want to highlight the `PYTHONPATH` warnings because I don't set this env variable so it must be homebrew and it's apparently setting a wrong one? Can this be? I really don't know much about brew's internals but this problem here could be explained by brew selecting a wrong python version imo.

Any ideas? Does it work for anyone else? :/

**Comment by Argon-** [comment]
I did some printf debugging and while `python_executable` is set correctly to `/usr/local/opt/python3/Frameworks/Python.framework/Versions/3.6/bin/python3` and `py_site_packages` is `/usr/local/Cellar/vtk/8.0.1_1/lib/python/site-packages` (that correct?), the other values are empty or wrong due to failed invocations. 
`python_prefix` and `python_include` are an empty string, `python_version` is `python` (due to "python" + empty string).

Each python invocation spawns one `PYTHONPATH` error as can be seen above. I have no idea where this env var comes from though.

**Comment by fxcoudert** [comment]
Can you tell us the output of `brew doctor`?

Something I don't understand: in the report, you say that L69 gives the correct path. But in the comment above, you say `python_prefix` (which is set at this same line) is empty.

**Comment by Argon-** [comment]
Brew doctor is clean except of two header files it complains about but they are unrelated and placed there by me intentionally. I'm currently not at my computer but I can supply you with the whole output if interested. 

I'm sorry, the first post must've been misleading. I executed these respective commands in my terminal by hand, to check what it outputs as I assumed these commands worked but just returned wrong values (eg in case my python setup was botched). 
In my second post I made the formula print the actual content of these variables.

**Comment by Argon-** [comment]
I just updated brew (last time was right before I opened this issue) and now it works. I strongly assume it is related to this commit but I'm only guessing: https://github.com/Homebrew/brew/pull/3577

**Comment by bastula** [comment]
I just tried this with Python 3 this weekend and ran into the same issue. It works fine linking with Python 2.

Here is my gist log: https://gist.github.com/bastula/997fdf989850bd991a4a94e2e77db389

**Comment by Argon-** [comment]
Are you sure it's the same problem? Asking because the commit linked above fixed this problem for me on multiple machines.
In case it's the same I can reopen this issue for you.

**Comment by bastula** [comment]
I will try tonight on a different machine to see if I can reproduce.
Building gcc takes forever 😢

**Comment by bastula** [comment]
Hmm it worked on a different machine. Granted I installed xcode tools so it had the bottled version of GCC. Let me try again with my original machine and install xcode tools to see if it helps.

**Comment by bastula** [comment]
Found the culprit. I was in a virtualenv on my other machine. Sorry for any confusion. 😊  hope this helps out anyone else who runs into this! 😆
